apps/frontend/src/store/blockStore.ts

```diff
@@ -22,3 +24,16 @@ export const removeUserFromBlockList = (userId: string) => {
 };
 
 export const userIsBlocked = (userId: string) => blocklist.value.includes(userId);
+
+export const iAmBlocked = async (location: string, userId: string): Promise<boolean> => {
+    const url = `http://[${location}]/api/v2/blocked/${userId}`;
+    return (await axios.get(calcExternalResourceLink(url))).data;
+};
+
+export const checkIfBlocked = async (contacts: GroupContact[], userId: string) => {
```

Comment:
this is not really checking if a contact is blocked rather filtering out blocked contacts.
I would change this to `getUnblockedContacts`